test/content-routing.node.js

```diff
@@ -7,87 +7,363 @@ const chai = require('chai')
+            protocol: 'http',
+            port: 60197
+          }, [
+            ma('/ip4/0.0.0.0/tcp/60194')
+          ])
+          nodeA = new Node({
+            peerInfo,
+            modules: {
+              contentRouting: [ delegate ]
+            },
+            config: {
+              relay: {
+                enabled: true,
+                hop: {
+                  enabled: true,
+                  active: false
+                }
+              },
+              EXPERIMENTAL: {
+                dht: true
+              }
+            }
+          })
+          nodeA.start(cb)
+        }
+      ], done)
     })
 
-    it('nodeE.contentRouting.findProviders for existing record', (done) => {
-      nodeE.contentRouting.findProviders(cid, 5000, (err, providers) => {
-        expect(err).to.not.exist()
-        expect(providers).to.have.length.above(0)
-        done()
+    after((done) => nodeA.stop(done))
+
+    describe('provide', () => {
+      it('should use both the dht and delegate router to provide', (done) => {
+        const dhtStub = sinon.stub(nodeA._dht, 'provide').callsFake(() => {})
+        const delegateStub = sinon.stub(delegate, 'provide').callsFake(() => {
+          expect(dhtStub.calledOnce).to.equal(true)
+          expect(delegateStub.calledOnce).to.equal(true)
+          delegateStub.restore()
+          dhtStub.restore()
+          done()
+        })
+        nodeA.contentRouting.provide()
       })
     })
 
-    it('nodeC.contentRouting.findProviders for non existing record (timeout)', (done) => {
-      const cid = new CID('QmTp9VkYvnHyrqKQuFPiuZkiX9gPcqj6x5LJ1rmWuSnnnn')
+    describe('findProviders', () => {
+      it('should only use the dht if it finds providers', (done) => {
+        const results = [true]
+        const dhtStub = sinon.stub(nodeA._dht, 'findProviders').callsArgWith(2, null, results)
+        const delegateStub = sinon.stub(delegate, 'findProviders').throws(() => {
+          return new Error('the delegate should not have been called')
+        })
 
-      nodeE.contentRouting.findProviders(cid, 5000, (err, providers) => {
-        expect(err).to.not.exist()
-        expect(providers).to.have.length(0)
-        done()
+        nodeA.contentRouting.findProviders('a cid', 5000, (err, results) => {
+          expect(err).to.not.exist()
+          expect(results).to.equal(results)
+          expect(dhtStub.calledOnce).to.equal(true)
+          expect(delegateStub.notCalled).to.equal(true)
+          delegateStub.restore()
+          dhtStub.restore()
+          done()
+        })
+      })
+
+      it('should use the delegate if the dht fails to find providers', (done) => {
+        const results = [true]
+        const dhtStub = sinon.stub(nodeA._dht, 'findProviders').callsArgWith(2, null, [])
+        const delegateStub = sinon.stub(delegate, 'findProviders').callsArgWith(2, null, results)
+
+        nodeA.contentRouting.findProviders('a cid', 5000, (err, results) => {
```

Comment:
Use the`maxTimeout` instead of `5000`, as we intend to deprecate it in the future.